Src/Common/WebHeaderCollectionExtensions.cs

```diff
@@ -19,10 +22,7 @@ internal static class WebHeaderCollectionExtensions
         /// <returns>Value against the given parameters.</returns>
         public static string GetNameValueHeaderValue(this NameValueCollection headers, string headerName, string keyName)
         {
-            if (string.IsNullOrEmpty(headerName))
-            {
-                throw new ArgumentNullException(nameof(headerName));
-            }
+            Debug.Assert(headerName != null, "headerName must not be null");
```

Comment:
```
       /// <summary>
        /// For the given header collection, for a given header of name-value type, find the value of a particular key.
        /// </summary>
        /// <param name="headers">Header collection.</param>
        /// <param name="headerName">Name of the header in the collection.</param>
        /// <param name="keyName">Desired key of the key-value list.</param>
        /// <returns>Value against the given parameters.</returns>
        public static string GetNameValueHeaderValue(this NameValueCollection headers, string headerName, string keyName)
        {
            Debug.Assert(headerName != null, "headerName must not be null");

            if (string.IsNullOrEmpty(keyName))
            {
                throw new ArgumentNullException(nameof(keyName));
            }

            var requiredHeader = headers[headerName];

```

There is one more throw there, I'll replace it with Debug.Assert as well. It could only happen if developer passed null by mistake